lib/pure/sugar.nim

```diff
@@ -66,12 +63,12 @@ macro `=>`*(p, b: untyped): untyped =
 
     type
       Bot = object
-        call: proc (): string {.nosideEffect.}
+        call: proc (name: string): string {.noSideEffect.}
 
     var myBot = Bot()
 
-    myBot.call = () {.nosideEffect.} => "I'm a bot."
```

Comment:
Please move original example(or both) to tests.

Reply:
Why?

Reply:
The PR removes the original example, add that to test in case of regression.

Reply:
Ah ok, thanks for clarifying.